### ISSUE TITLE
SDMX-JSON with time dimension can fail in master

### ISSUE THREAD

**Issue by RacingTadpole** [TerriaJS/terriajs#2723]
This line of `CatalogItemDateTimePicker`:
```
var currentDate = availableDates[item.intervals.indexOf(currentTime)];
```
can fail with `item.intervals` undefined when you change the selected concept.

**Comment by kring** [comment]
Is this a duplicate of #2793 and fixed by #2811?  @steve9164 can you confirm?

**Comment by tephenavies** [comment]
Should be fixed. Each place where `CatalogItemDatePicker` is conditionally rendered based on existence of `item.intervals` is now properly updated using ObserveModelMixin. I.e. `CatalogItemDatePicker` should never render when `item.intervals` is undefined/null.
